test/compiler/inference.jl

```diff
@@ -1622,24 +1622,24 @@ let memoryref_tfunc(@nospecialize xs...) = Core.Compiler.memoryref_tfunc(Core.Co
     @test memoryref_isassigned_tfunc(Any, Any, Any) === Bool
     @test builtin_tfunction(Core.memoryref_isassigned, Any[MemoryRef{Int}, Vararg{Any}]) == Bool
     @test builtin_tfunction(Core.memoryref_isassigned, Any[MemoryRef{Int}, Symbol, Bool, Vararg{Bool}]) == Bool
-    @test memoryrefset!_tfunc(MemoryRef{Int}, Int, Symbol, Bool) === MemoryRef{Int}
```

Comment:
wait, why are all of these changing the return type?

Reply:
It was inconsistent with all the other similar builtins, which made for a rough time when trying to write the codegen for them. Better for them all to behave the same, and these were not public currently or in a release yet, so easier to change them now.

Reply:
fair enough.